frontend/src/components/CourseList.tsx

```diff
@@ -1,120 +1,49 @@
 import React, { useEffect } from "react";
 import { useAppDispatch, useAppSelector } from "../app/hooks";
 import CourseCard from "./CourseCard";
-import { Pagination } from "react-headless-pagination";
 import {
-  fetchCourseInfosByPage,
+  fetchCourseInfos,
   fetchFCEInfos,
   selectCourseResults,
 } from "../app/courses";
-import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/outline";
 import Loading from "./Loading";
 
-const CoursePage = () => {
-  const dispatch = useAppDispatch();
-
-  const pageCourses = useAppSelector((state) => state.courses.pageCourses);
-  const page = useAppSelector((state) => state.courses.page);
-
-  const exactResultsCourses = useAppSelector(
-    (state) => state.courses.exactResultsCourses
-  );
-  const exactMatchesOnly = useAppSelector(
-    (state) => state.user.filter.exactMatchesOnly
-  );
+interface Props {
+  courseIDs: string[];
+  children: React.ReactNode;
+}
 
-  const showFCEs = useAppSelector((state) => state.user.showFCEs);
-  const showCourseInfos = useAppSelector((state) => state.user.showCourseInfos);
+const CourseList = ({ courseIDs, children }: Props) => {
   const loggedIn = useAppSelector((state) => state.user.loggedIn);
-
-  let coursesToShow: string[] = [];
-
-  if (exactMatchesOnly) {
-    coursesToShow = [...exactResultsCourses];
-  } else if (page === 1 && exactResultsCourses.length > 0) {
-    coursesToShow = [
-      ...exactResultsCourses,
-      ...pageCourses.filter(
-        (courseID) => !exactResultsCourses.includes(courseID)
-      ),
-    ];
-  } else {
-    coursesToShow = [...pageCourses];
-  }
-
-  const results = useAppSelector(selectCourseResults(coursesToShow));
-  const coursesToShowStr = coursesToShow.join(" ");
+  const loading = useAppSelector((state) => state.courses.coursesLoading);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
-    if (loggedIn && coursesToShow) {
-      dispatch(fetchFCEInfos({ courseIDs: coursesToShow }));
+    if (courseIDs) {
+      dispatch(fetchCourseInfos(courseIDs));
+      if (loggedIn) dispatch(fetchFCEInfos({ courseIDs }));
     }
-  }, [coursesToShowStr, loggedIn]);
+  }, [loggedIn]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'courseIDs' and 'dispatch'. Either include them or remove the dependency array.](https://app.codacy.com/gh/ScottyLabs/course-tool/pullRequest?prid=9667658)
```suggestion
  }, [courseIDs, dispatch, loggedIn]);
```